docker/coreboot/Dockerfile

```diff
@@ -1,11 +1,14 @@
 # This is multi-stage Dockerfile
 #   Docs: https://docs.docker.com/build/building/multi-stage/
 
+ARG TARGETARCH=amd64
+
 #=============
 # "base" stage with all needed build dependencies
 FROM ubuntu:jammy AS base
 
-ARG TARGETARCH=amd64
+ARG TARGETARCH
```

Comment:
Can you briefly explain why we define TARGETARCH twice, once with a value (like previously) and now a second time without one?

Reply:
Oh yeah, sorry.

It is because of the `ARG` limited [scope](https://docs.docker.com/reference/dockerfile/#scope).

> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the ARG instruction.

TLDR: `FROM` starts a new scope (a new stage) where the previous `ARG` does not apply anymore. It is possible to use `ARG` without value if the previous `ARG` was declared in the top scope (aka before first `FROM`).